src/Essentials/src/FilePicker/FilePicker.uwp.cs

```diff
@@ -42,13 +43,52 @@ async Task<IEnumerable<FileResult>> PlatformPickAsync(PickOptions options, bool
 
 			if (AppInfoUtils.IsPackagedApp)
 			{
-				foreach (var file in resultList)
-					StorageApplicationPermissions.FutureAccessList.Add(file);
+				AddToAndCleanUpFutureAccessList(resultList);
 			}
 
 			return resultList.Select(storageFile => new FileResult(storageFile));
 		}
 
+		void AddToAndCleanUpFutureAccessList(List<StorageFile> pickedFiles)
+		{
+			var fal = StorageApplicationPermissions.FutureAccessList;
+
+			try
+			{
+				// Check if (FutureAccessList current items + new picked files) is exceeding maximum items
+				if ((fal.Entries.Count + pickedFiles.Count) >= fal.MaximumItemsAllowed)
+				{
+					// We assume that the FutureAccessList items are saved in order, there is no metadata
+					// Therefore we start removing from the bottom of the list
+					var removeCount = Math.Min(fal.Entries.Count, pickedFiles.Count);
```

Comment:
Something is bothering me about the math here... Will revisit in the morning...

Reply:
Yes, please! Double-check it. Yesterday this made sense, but I agree it can seem a bit weird-ish now. Let me know your thoughts, I might have a second look myself.

Reply:
So I think the reason why I added it like this was: the current access list entries were less then the files we picked right now. This only happens when the picked files are more than the maximum capacity of the access list, usually 1000. 

Imagine that the access list has 1000 entries already and we pick 1001 files. If I would just assume we need to clear the amount of places in the access list for the files that are picked, I was going to remove 1001 files, while there are only 1000 entries in the access list. BOOM!

That's why I added this condition, but it should almost never be hit @mattleibow